addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3523,6 +3523,13 @@ var SnippetsMenu = Widget.extend({
         }
 
         var srcElement = ev.target || (ev.originalEvent && (ev.originalEvent.target || ev.originalEvent.originalTarget)) || ev.srcElement;
+        const selection = this.options.wysiwyg.odooEditor.document.getSelection();
+        if (srcElement.querySelector('.oe_unremovable')) {
+            const parent = selection.anchorNode && selection.anchorNode.parentElement
+            if(parent){
+                srcElement = parent;
+            }
+        }
```

Comment:
I don't understand why this file would need to be changed. I suppose this is related to the colorpicker flickering ? Or maybe it's the toolbar disappearing ? I can't know easily from the changeset because you fix two different bugs in one commit. Could you please split the changes in two commit so that it is easier to understand what change is supposed to affect what ? Thanks !

In any case, I do not understand the relation between either of the bugs and the fact that the node might be unremovable. Why does being unremovable play any role here ?

Reply:
This is related to the toolbar disappearing.
In this case when we move from right to left with a selection then we also select and other div that is present above that text since the cursor is still present inside the editable area and in this case we get a div with .oe-unremovable class in our event target. So, I had just checked that if target is unremovable and we have selection present in our editable area then reassign srcElement with the selection. This will stop the tab to change from 'OPTIONS' to 'BLOCK'

Reply:
@dmo-odoo Please look into this. Thank You!

Reply:
Sorry for the late reply. I feel like this issue is not related to the node being unremovable. I think the node might be unremovable yes but the same issue could happen with a removable node, don't you think ?

Reply:
You are right that issue is not related to the node being `removable` or `unremovable`.
When we do selection starting from text and continue selecting more content and as we go up we get div in our selection. 
And as we get a div in our selection we get whole editable area in our `ev.target`,  this causes the Customize Tab to shift to Block Tab. So, to stop the sifting of the tab I just checked if the `ev.target` has the `unremovabel` class then reassign the target text so that the tab remains to customize tab.

Reply:
You are the issue the issue is not related to the node being removable or unremovable. 
Instead, when selecting content starting from text and continuing to include additional content upwards, which introduces a div element into the selection range causes the shifting of the Customize Tab to the Block Tab.
As a consequence, the ev.target element encompasses the entire editable area, leading to the undesired shift to the Block Tab. By performing a check to determine if the ev.target element possesses the "unremovable" class, we can reassign the ev.target to text element which prevents the unintended shifting of the Tab

Reply:
I wanted to suggest you to look for the `editable` root node itself rather than looking for the unremovable class but then I realized we're in fact in `snippets.editor.js` and not in the editor per se. Checking for the unremovable class is not the right fix IMO, it just _happens_ that the root is unremovable but that isn't really what we want. Would it be possible to make a similar check in Odoo Editor where you can look that the node is in fact the editable root one so that you don't have to rely on checking for the unremovable class ?

Reply:
@dmo-odoo
As you mentioned, I tried checking for the editable root node, and initially, the problem seemed to be solved. However, I encountered an issue when attempting to intentionally move to the `BLOCK` tab while keeping the cursor inside the editable area. In this scenario, I found that I was unable to switch to the block tab because the editable root node was still being detected, thus preventing the changing of tab. Additionally instead of 
`srcElement.querySelector('.oe_unremovable')` I tried this
`srcElement.classList.contains('o_editable') || srcElement.querySelector('.o_editable')` and it was working fine.

Reply:
Ok great, thanks ! 👍 Now, I suggest:
- Move the `const selection` inside of the `if` as it is not used outside of it.
- Add a space between the `if` and the parenthesis for the second `if`.
- Write a comment before the first `if` to explain the problem that happens when `srcElement` is not contained inside of the editable and why the code inside the if actually fixes the bug.

Reply:
@dmo-odoo Hello, 
I have changes the `o_editable` class to `o_mail_wrapper` because it was breaking a tour from website during my last code update. Other than that I have done all the changes you suggested. Thank You!!

Reply:
@maad-odoo Sorry I just noticed now that a comment I wrote on July 13 had not been published properly. It is the comment above.

Reply:
@dmo-odoo I just updated the code and tried a different approach. Would you please take a look at it and suggest whether this approach seems correct to you or not?

Reply:
@dmo-odoo A gentle reminder.

Reply:
@maad-odoo This seems to be better, thanks ! However, that's a big change compared to the original proposed solution so I think it would be better to go through testing in India again if you don't mind.